lib/ansible/module_utils/basic.py

```diff
@@ -902,8 +902,9 @@ def selinux_context(self, path):
         context = ret[1].split(':', 3)
         return context
 
-    def user_and_group(self, filename):
-        filename = os.path.expanduser(os.path.expandvars(filename))
+    def user_and_group(self, filename, expand=True):
+        if expand:
+            filename = os.path.expanduser(os.path.expandvars(filename))
         b_filename = to_bytes(filename, errors='surrogate_or_strict')
```

Comment:
from my testing we should switch the order of these:
``` python
b_filename = to_bytes(filename, errors='surrogate_then_strict')
if expand:
    b_filename = os.path.expanduser(os.path.expandvars(b_filename))
# Optional: only needed if the method uses filename instead of b_filename later
filename = to_text(b_filename, errors='surrogate_then_strict')
```